Subject: customize, resize, sparsify, sysprep: Link explicitly with pthread
From: Richard W.M. Jones rjones@redhat.com Thu Mar 4 11:57:44 2021 +0000
Date: Sat Sep 11 09:42:12 2021 +0100:
Git: 68a02c2f6c7b4243ecb298c0d9539f0fc51a52ce

Cherry picked from guestfs-tools commit 87543dad61.

diff --git a/customize/Makefile.am b/customize/Makefile.am
index c926687b3..fb418d7bf 100644
--- a/customize/Makefile.am
+++ b/customize/Makefile.am
@@ -169,6 +169,7 @@ OCAMLLINKFLAGS = \
 	$(LINK_CUSTOM_OCAMLC_ONLY)
 
 OCAMLCLIBS = \
+	-pthread -lpthread \
 	-lutils \
 	$(LIBTINFO_LIBS) \
 	$(LIBCRYPT_LIBS) \
diff --git a/resize/Makefile.am b/resize/Makefile.am
index fa88cc634..936d2b05e 100644
--- a/resize/Makefile.am
+++ b/resize/Makefile.am
@@ -72,6 +72,7 @@ OCAMLPACKAGES += -package gettext-stub
 endif
 
 OCAMLCLIBS = \
+	-pthread -lpthread \
 	-lprogress \
 	-lutils \
 	$(LIBTINFO_LIBS) \
diff --git a/sparsify/Makefile.am b/sparsify/Makefile.am
index fa964e8af..4d1af85d6 100644
--- a/sparsify/Makefile.am
+++ b/sparsify/Makefile.am
@@ -80,6 +80,7 @@ OCAMLPACKAGES += -package gettext-stub
 endif
 
 OCAMLCLIBS = \
+	-pthread -lpthread \
 	-lprogress \
 	-lutils \
 	$(LIBTINFO_LIBS) \
diff --git a/sysprep/Makefile.am b/sysprep/Makefile.am
index 69bb92820..250c74913 100644
--- a/sysprep/Makefile.am
+++ b/sysprep/Makefile.am
@@ -129,6 +129,7 @@ OCAMLPACKAGES += -package gettext-stub
 endif
 
 OCAMLCLIBS = \
+	-pthread -lpthread \
 	-lvisit \
 	-lstructs \
 	-lutils \
